nix/packages/treefmt/package.nix

```diff
@@ -0,0 +1,49 @@
+{
+  buildGoModule,
+  installShellFiles,
+  lib,
+  pname,
+  src,
+  version,
+  passthru,
```

Comment:
This was the funkiest part of this refactor. We seem to be using `passthru` for 2 things:

1. Stash a coverage test somewhere so `.github/workflows/coverage.yml` can invoke it with `nix build .#treefmt.tests.coverage`.
2. Stash the update vendor hash utility somewhere so `.github/workflows/update-vendor-hash.yml` can invoke it with `nix run .#treefmt.update-vendor-hash`.

To minimize our diff with nixpkgs, I didn't want leave these in `package.nix`, so I opted to pass them through like this. I'm very open to alternate ideas for how to tackle this, though.